### PR TITLE
fix: `SUMMARY_MAX_PARAGRAPHS` not respected in some combinations with `SUMMARY_MAX_LENGTH`

### PR DESCRIPTION
**Pull request by scolby33**
When, for example, the content is 2 paragraphs, each consisting of 30 words, and `SUMMARY_MAX_PARAGRAPHS=1` and `SUMMARY_MAX_LENGTH=50`, the current behavior produces a summary with 50 words and 2 paragraphs, effectively ignoring `SUMMARY_MAX_PARAGRAPHS`.

This change inverts this behavior: the resulting summary from the above situation will consist of the entire first 30 word paragraph.

In the case where the first paragraph(s) is longer than `SUMMARY_MAX_LENGTH`, the summary will still be truncated to that length.

This change brings the actual behavior in line with the documented behavior.

# Pull Request Checklist

- [x] Ensured **tests pass** and (if applicable) updated functional test output
- [x] Conformed to **code style guidelines** by running appropriate linting tools
- [x] Added **tests** for changed code
- [x] ~Updated **documentation** for changed code~ no need for documentation changes

### CODE REVIEW

File: pelican/tests/test_contents.py [hunk 1]
```diff
@@ -14,7 +14,9 @@
 from pelican.utils import path_to_url, posixize_path, truncate_html_words
 
 # generate one paragraph, enclosed with <p>
```
Comment (cthoyt) [inline pelican/tests/test_contents.py:16]: need to update this comment

Comment (scolby33) [reply pelican/tests/test_contents.py]: Thanks!

**Comment by justinmayer** [comment]
Thank you, Scott, for submitting this pull request — replete with a test, no less! ✨

Given the narrow scope of these changes, I think a release type of `patch` would be more appropriate. Would you please change the release file accordingly?

@gagath: As the original author of the `SUMMARY_MAX_PARAGRAPHS` setting, do you have any comments on this PR?

**Comment by scolby33** [comment]
@justinmayer no problem, changed the release type in the latest version!

**Review by justinmayer — APPROVED** [review]
Many thanks to @scolby33 for the enhancement and to @cthoyt & @gagath for reviewing! 🏅

**Comment by cthoyt** [comment]
@justinmayer not relevant to the PR but @scolby33 and I are flatmates and we thought it was funny we both had random PRs for two of your different projects back to back. Cheers!

**Comment by justinmayer** [comment]
No way!  😮  That is indeed quite funny — literally laughed out loud when I read it.

It’s been a pleasure. I hope there will be more random open-source encounters with y’all in the future. If you’re ever in the Italian Alps, beers are on me! 🍻

**Pull request merged**